src/AppInstallerRepositoryCore/CompositeSource.cpp

```diff
@@ -436,92 +436,105 @@ namespace AppInstaller::Repository
     {
         CompositeResult result;
 
-        // Search installed source
-        SearchResult installedResult = m_installedSource->Search(request);
-        result.Truncated = installedResult.Truncated;
-
-        for (auto&& match : installedResult.Matches)
+        // If the search behavior is for AllPackages or Installed then the result can contain packages that are
+        // only in the Installed source, but do not have an AvailableVersion.
+        // If the search behavior is for AvailablePackages, then most queries can be resolved by just querying the available source
+        // and then correlating it with the installed source, but if the query is for everything then query the installed catalog
+        // and only add installed packages that have an AvailableVersion.

```

Comment:
I don't think that this makes sense, if you ask for Available only, it should not go down this path at all.

Reply:
It's consistent with the early return when request.IsForEverything() and search behavior is AllPackages. Thought that was by design. The feature of "get me everything that's currently installed from the available catalogs" is good and needed somehow.

Reply:
I don't think that was thought through; the search behavior has been tacked onto the original design and was only ever intended for use with import.  In that case it was never for everything.

I think it is better to make it make sense now than try to preserve an accidental behavior.